src/pyhf/infer/intervals.py

```diff
@@ -15,7 +22,132 @@ def _interp(x, xp, fp):
     return tb.astensor(np.interp(x, xp.tolist(), fp.tolist()))
 
 
-def upperlimit(data, model, scan, level=0.05, return_results=False, **hypotest_kwargs):
+def upperlimit_auto(
+    data,
+    model,
+    low,
+    high,
+    level=0.05,
+    atol=2e-12,
+    rtol=None,
+    calctype="asymptotics",
+    test_stat="qtilde",
+    from_upperlimit_fn=False,
```

Comment:
These should probably be taking the hypotest_kwargs like `upperlimit_fixedscan` does at the moment. Might want to do so renaming of things as well.